dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -419,57 +442,111 @@ def _external_source(self, name=None, **kwargs):
             raise RuntimeError(f"Unexpected operator 'ExternalSource'. Debug mode does not support"
                                " changing the order of operators executed within the pipeline.")
 
-    def _run_subpipeline(self, pipe_tuple, op_name, inputs, kwargs):
-        """Run pipeline wrapper of a single operator."""
-        self._debug_on = False
-        pipe, inputs_es, kwargs_es = pipe_tuple
+    def _run_op_on_device(self, op_name, device, inputs, kwargs):
+        if device == 'gpu':
+            return self._pipe.RunOperatorGPU(self._cur_op_id, inputs, kwargs)
+        if device == 'cpu':
+            return self._pipe.RunOperatorCPU(self._cur_op_id, inputs, kwargs)
+        if device == 'mixed':
+            return self._pipe.RunOperatorMixed(self._cur_op_id, inputs, kwargs)
 
-        if len(inputs_es) != len(inputs):
-            raise RuntimeError(f"Trying to use operator '{op_name}' with different number of inputs than when"
-                               f" it was built. {len(inputs_es)} != {len(inputs)}")
-        if len(kwargs_es) != len(kwargs.items()):
-            raise RuntimeError(f"Trying to use operator '{op_name}' with different number of keyward arguments"
-                               " than when it was built.")
+        raise ValueError(f"Unknown device: '{device}' in operator '{op_name}'.")
 
-        def unexpected_argument_msg(to_external_source):
-            return f"recognized as {'batch' if to_external_source else 'constant'} but when built value" \
-                f" in its place was recognized as {'constant' if to_external_source else 'batch'}"
+    def _pack_to_data_node_debug(self, data, op_name):
+        if isinstance(data, (list, tuple)):
+            return [self._pack_to_data_node_debug(elem, op_name) for elem in data]
 
+        return DataNodeDebug(data, op_name, 'gpu' if isinstance(data, _tensors.TensorListGPU) else 'cpu', self)
+
+    def _run_op(self, op_tuple, op_name, inputs, kwargs):
+        """Run a single operator."""
+
+        def check_arg_len(expected_len, actual_len, args_type):
+            if expected_len != actual_len:
+                raise RuntimeError(f"Trying to use operator '{op_name}' with different number of {args_type} than when"
+                                   f" it was built. Expected: {expected_len} {args_type}, got {actual_len}.")
+
+        def check_classification(expected_is_batch, actual_is_batch, arg_type, value):
+            def classification_to_str(is_batch):
+                return 'batch' if is_batch else 'constant'
+
+            if expected_is_batch != actual_is_batch:
+                expected_str = classification_to_str(expected_is_batch)
+                actual_str = classification_to_str(actual_is_batch)
+
+                raise RuntimeError(f"In operator '{op_name}' {arg_type} {value} recognized as {actual_str} but built value"
+                                   f" in its place was recognized as {expected_str}.")
```

Comment:
```suggestion
                raise RuntimeError(f"The operator '{op_name}' {arg_type} {value} recognized as {actual_str} but it was created"
                                   f" with {expected_str}.")
```

Reply:
done

Reply:
Were you thinking of e.g. `The operator 'random__CoinFlip' recognized argument 'probability' as batch but it was created with constant.`?
Because this gives `The operator 'random__CoinFlip' argument probability recognized as batch but it was created with constant.`

Reply:
That sounds better. The original just sounds strange for me.